File: Python/14.py
class Solution:
    # @param {string[]} strs
    # @return {string}
    
    #Version1
    def longestCommonPrefix(self, strs):
        if not strs:
            return ""
        com = strs[0]
        for i in strs:
            com = com[:min(len(i),len(com))]    
            for k in range(len(com)):
                if com[k] != i[k]:
                    com = com[:k]
                    break
        return com


    """ Version 2 form others
    def longestCommonPrefix(self, strs):
        if not strs:
            return ''
        str = strs[0]
        res = ''
        for i in range(len(str)):
            for j in range(1,len(strs)):
                if i > len(strs[j]) - 1:
                    return res
                if str[i] != strs[j][i]:
                    return res
            res += str[i]
        return res
    """

File: C++/14.cpp
class Solution {
public:
/* Version 1
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        string common = strs[0];
        for(int i = 1;i<strs.size();i++){
            for(int j=0; j<common.size();j++) if(common[j]!=strs[i][j]) common = common.substr(0,j);
            if(common.empty()) return "";
        }
        return common;
    }
*/
// Version 2
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        string common;
        for(int i = 0;i<strs[0].size();i++){
            for(int j=0; j<strs.size();j++) if(strs[0][i]!=strs[j][i]) return common;
            common += strs[0][i];
        }
        return common;
    }
};